frundenbot/main.py

```diff
@@ -38,13 +44,48 @@
 
 LOGGER = logging.getLogger(__name__)
 
-TOKEN = os.environ.get('TELEGRAM_BOT_AUTH_TOKEN')
 ADMIN = int(os.environ.get('TELEGRAM_BOT_ADMIN'))
 LIST_OF_ADMINS = [ int(x) for x in os.environ.get('TELEGRAM_BOT_ADMINS').split(',') ]
 
-LOG_TIME = Summary('frunde_log_seconds', 'Time spent logging incomming message')
-@LOG_TIME.time()
-def __log_incomming_messages(bot, update):
+class FrundenBot:
+    def __init__(self, token, refresh_interval):
+
+        self.FRUNDE_OPEN = Gauge('frunde_status', '1 if Frunde is open,-1 on error, 0 otherwise')
+
+        updater = Updater(token=token, use_context=True)
+        dispatcher = updater.dispatcher
+        queue = updater.job_queue
+        queue.run_repeating(self.refresh_cache, interval=refresh_interval, first=0)
+        me = dispatcher.bot.get_me()
+        LOGGER.info('Running as %s (%s)', me.username, me.id)
+
+        handler_groups = {
+            0: [MessageHandler(None, callback=self._callback_log_message)],
+            1: [
+                InlineQueryHandler(callback=self._callback_inline),
+                CommandHandler('commands', callback=self._callback_commands),
+                CommandHandler('start', callback=self._callback_start),
+                CommandHandler('open', callback=self._callback_is_open),
```

Comment:
I think it is possible to pass a list of commands here, so you dont have to add loads of command handlers when they all use the same callback.